src/lib/db.ts

```diff
@@ -46,6 +58,8 @@ export class DB {
   private groupOnDisconnect: firebase.database.OnDisconnect | null = null;
   private connectedRef: firebase.database.Reference | null = null;
   private workspaceRef: firebase.database.Reference | null  = null;
+  private userDocumentRefs: ReferenceMap = {};
+  private userDocumentDisposers: DisposerMap = {};
```

Comment:
I think it might be cleaner if you defined an interface that had both the reference and the disposer as members and then define a map on that interface.  Then you could have a single variable since both maps are parallel.

Reply:
I gave this a try - ended up adding a bunch of null checks, let me know if you had something else in mind.